core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala

```diff
@@ -328,6 +328,14 @@ object SparkSubmit {
       }
     }
 
+    // In yarn mode for a python app, if PYSPARK_ARCHIVES_PATH is in the user environment
+    // add pyspark archives to files that can be distributed with the job
+    if (args.isPython && clusterManager == YARN){
+      sys.env.get("PYSPARK_ARCHIVES_PATH").map { archives =>
+        args.files = mergeFileLists(args.files, Utils.resolveURIs(archives))
```

Comment:
I think you have to put them on `args.pyFiles`. Otherwise they won't be added to the executors' `PYTHONPATH`s